### ISSUE TITLE
Relative reference to Bootstrap in main.scss

### ISSUE THREAD

**Issue by markthethomas** [yeoman/generator-webapp_DEPRECATED#388]
On `grunt serve` this error is thrown:

`Warning: Syntax error: File to import not found or unreadable: ../../bower_components/bootstrap-sass-official/vendor/assets/stylesheets/bootstrap.scss.
                  Load paths:
                    /Users/mthomas/Code/apps/testing (DEPRECATED)
                    /Users/mthomas/Code/apps/testing/app/styles
                    /Users/mthomas/Code/apps/testing/bower_components
            on line 4 of app/styles/main.scss
      Use --trace for backtrace. Use --force to continue.`

A simple fix is the making sure the relative reference doesn't go deeper than it should:
`@import '../../bower_components/bootstrap-sass-official/...`
to 
`@import '../bower_components/bootstrap-sass-official/...`

This might be intentional and related to a patch/fix I'm not aware of, but wanted to flag it just in case & make before submitting a PR.

**Comment by climboid** [comment]
Thanks for that fix @markthethomas

**Comment by markthethomas** [comment]
No prob; can a maintainer let me know if this is worth a PR or if it can be squashed into another one? (Seems small enough to be included in another more important PR)

**Comment by silvenon** [comment]
Which version are you using? 0.4.9 or master?

**Comment by patrickbjohnson** [comment]
Was there any actual resolution for this? Just updated all generators (using yeoman webapp but still running into an error. Fails 100% of the time and matching relative paths seems to not fix the issue.

**Comment by eddiemonge** [comment]
what version?

**Comment by patrickbjohnson** [comment]
webapp/yeoman - v. 1.2.0

ran `yo webapp -v` and that was the result, but noticed that running `yo -v` results to the same version

**Comment by silvenon** [comment]
Make sure your Bootstrap path is correct, because in version 3.2 Bootstrap changed its directory structure, so instead of `.../vendor/assets/...` you should have `.../assets/...`. It's a long shot, but check if that's the case.

**Comment by eddiemonge** [comment]
@patrickbjohnson run `yo` and it will say what version of webapp you have.

**Comment by madodela** [comment]
Thank's @silvenon, what you said helps me to figure out what was wrong in my project, because I had my folder structure like .../vendor/assets/... and the path in the import was .../assets/...

Thank you so much

**Comment by silvenon** [comment]
@madodela :+1:

**Comment by rreusser** [comment]
I'm running into an extremely slight variation of this same issue with v1.2.1.  I get:

```
Warning: Syntax error: File to import not found or unreadable: bootstrap-sass-official/assets/stylesheets/_bootstrap.scss.
              Load paths:
                /Users/username/projects/project-name (DEPRECATED)
                /Users/username/projects/project-name/bower_components
        on line 3 of app/styles/main.scss
```

Editing main.scss has no effect since `grunt serve` overwrites changes to sass `@import` paths. My temporary solution is, in the root project directory:

```
$ ln -s app/bower_components bower_components
```

Since:
1. I don't know how to send it to `project-name/app/bower_components`
2. I just want to make a webapp

**Comment by silvenon** [comment]
Your `bower_components` directory _should_ be in your project root, not in the `app` directory. Does this happen with a fresh install? Do you have a `.bowerrc` hidden in your project or maybe in your system root (`~`)?

**Comment by rreusser** [comment]
Good catch. `~/.bowerrc`:

```
{
    "directory": "app/bower_components"
}
```

I wonder who put that there. I'm (obviously) newish to the node/bower/yeoman world and was moderately surprised how difficult it was to use the built-in generators… I recall being very frustrated at the time of writing, but all good now.

tl;dr: Sorry to bother you! Thanks!

**Comment by silvenon** [comment]
No problem. I'm surprised by how many people have a `~/.bowerrc` because I don't think Bower has any documentation about putting it in the root.

**Comment by rreusser** [comment]
Some quick googling (automatic bowerrc creation? bower-rails gem?) didn't turn up much, but FWIW I don't have any recollection of adding this. The computer is six months old, primarily rails with a bit of python. Not much node other than yeoman. Perhaps there's a package that prefers (preferred?) this.

Since this is the first thing people do on the Getting Started with Yeoman page, it might not be totally out of line to add a warning if things are configured such that a fresh install will definitely fail. I have to confess I may have come to some hasty judgements about yeoman when it failed on a completely fresh (well…) node install. Everything since that has been pretty good though!

**Comment by silvenon** [comment]
Now the generator adds a local `.bowerrc`, hopefully there will be no more problems after this change is released.

**Comment by jamesongamble** [comment]
For some reason my app.scss file had a path of: 

@import 'bootstrap-sass-official/vendor/assets/stylesheets/bootstrap';
@import 'font-awesome/scss/font-awesome';

I had to prepend both lines with a ../../bower_components to make it run:

@import '../../bower_components/bootstrap-sass-official/vendor/assets/stylesheets/bootstrap';
@import '../../bower_components/font-awesome/scss/font-awesome';

Hope this helps.

**Comment by eddiemonge** [comment]
@jamesongamble what version are you running? It shouldn't need those since the include path is set in grunt: https://github.com/yeoman/generator-webapp/blob/master/app/templates/Gruntfile.js#L196

**Comment by Arieg419** [comment]
Hey,

While generating my app this error was thrown as well.

The poblem ended up being the firewall installed from the wifi I was connected to. 
As soon as I connected through a different network (unsecured) It worked fine.

Hope it helps, 

Cheers!

**Comment by FranzZemen** [comment]
I had the same problem then realized my .bowerrc wasn't setup.  It points to where bower should install bower_components
